src/lib/p4/github.py

```diff
@@ -124,3 +104,15 @@ def get_pr_test_instance(pr_endpoint, prefix='[Test Env] '):
             return label['name'][len(prefix):]
 
     return False
+
+def has_open_pr_labeled_with_instance(name):
```

Comment:
Isn't that better to be moved in the swarm dashboard? So when we also look the dashboard we know which instance are reserved.

https://github.com/greenpeace/planet4-test-swarm/pull/2

Reply:
Indeed. Can we do that in a second iteration so that we can already prevent the issues mentioned in the description?

Reply:
For the labels fix totally. That's a good enhancement. But the availability check seems like double the work, since the code will mostly be the same on the swam side and the PR is already progress.

Reply:
Absolutely. Yet it would require additional work to get it working there and I don't want to delay these fixes which would prevent people losing time. We can add it to the swarm and remove it here in a follow up.

Reply:
It's just a few lines of code, so it's indeed fine to iterate on this.

Reply:
There is actually a challenge to adding it to the swarm dashboard. The search endpoint is not very fast, which is not a problem if it's used in this job (adds max about 30 seconds in worst case of not a lot of instances assigned on Jira). With the current code we could even optimize this so that it orders the instances first and then does a `list(find(...))` instead of a `list(map(...))`.

If we move this to the swarm dashboard then it would need to do this for all instances all the time, which would significantly increase the response time of the dashboard page (could be 20/30 seconds). Perhaps it can be solved if we can send all requests in parallel.

Reply:
On swarm we also query the Jira API first, so we don't need to iterate through all instances, just the ones left. We do need the dashboard to be accurate, for cases where we want to manually pick a test instance. But if it slows down too much the `SWARM_API` call here, we can keep the check in both sides.